src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -265,6 +270,62 @@ public string RemoteDebugMachine
             }
         }
 
+        public IEnumerable<IRemoteAuthenticationProvider> RemoteAuthenticationProviders
+        {
+            get
+            {
+                var providers = new List<IRemoteAuthenticationProvider>(_authenticationProviders.ExtensionValues());
+                if (_unknownProvider != null)
+                {
+                    providers.Add(_unknownProvider);
+                }
+                return providers;
+            }
+        }
+
+        public IRemoteAuthenticationProvider RemoteAuthenticationProvider
+        {
+            get
+            {
+                Guid portSupplier = GetOtherProperty(LaunchProfileExtensions.RemoteAuthenticationPortSupplierProperty, Guid.Empty);
+                return GetProviderForPortSupplier(portSupplier);
+            }
+            set
+            {
+                if (TrySetOtherProperty(LaunchProfileExtensions.RemoteAuthenticationPortSupplierProperty, value.PortSupplier, Guid.Empty))
+                {
+                    OnPropertyChanged(nameof(RemoteAuthenticationProvider));
+                }
+            }
+        }
+
+        private IRemoteAuthenticationProvider GetProviderForPortSupplier(Guid portSupplier)
+        {
+            IRemoteAuthenticationProvider provider = _authenticationProviders.FirstOrDefaultValue(p => p.PortSupplier.Equals(portSupplier));
+
+            // If there is a bad guid in the launch settings we don't want to clobber it by having a default selection, so create a temporary provider
+            if (provider == null)
+            {
+                provider = _unknownProvider = new UnknownRemoteAuthenticationProvider(portSupplier);
```

Comment:
If I'm reading this corectly unknownProvider can either represent the thing the user just chose from the dialog if we don't recognize it, or the thing stored in launchSettings.json if we don't recognize it. Is there a way without needing state to store "unknown provider"?

Reply:
We can do this without the state, it just means the dropdown is blank and the user has to be careful not to change its value or they won't be able to get it back (and the property pages don't ask if you want to save when you close). The state means we can put this unknown provider in the dropdown itself so its easier to use. It does disappear once a valid selection is made though.

Having said that, I guess strictly speaking we don't need the state to provide the nicer functionality, it can just look up the property value and create another new UnknownRemoteAuthenticationProvider when constructing the list of providers if its not found.

Reply:
Removed this whole thing